src/audio/codec_adapter/codec_adapter.c

```diff
@@ -166,8 +166,19 @@ int codec_adapter_prepare(struct comp_dev *dev)
 	if (md->mpd.in_buff_size != mod->period_bytes)
 		mod->deep_buff_bytes = MIN(mod->period_bytes, md->mpd.in_buff_size) * buff_periods;
 
-	/* Allocate local buffer */
+	if (md->mpd.out_buff_size > mod->period_bytes) {
+		buff_periods = (md->mpd.out_buff_size % mod->period_bytes) ?
+			       (md->mpd.out_buff_size / mod->period_bytes) + 2 :
+			       (md->mpd.out_buff_size / mod->period_bytes) + 1;
+	} else {
+		buff_periods = (mod->period_bytes % md->mpd.out_buff_size) ?
+			       (mod->period_bytes / md->mpd.out_buff_size) + 2 :
+			       (mod->period_bytes / md->mpd.out_buff_size) + 1;
+	}
+
 	buff_size = MAX(mod->period_bytes, md->mpd.out_buff_size) * buff_periods;
+
+	/* Allocate local buffer */
 	if (mod->local_buff) {
 		ret = buffer_set_size(mod->local_buff, buff_size);
```

Comment:
so, `deep_buff_bytes` depends on the  `in_buff_size` and `period_bytes`, while local_buff_size depends on `period_bytes` and `out_buff_size`. 

Makes sense.